bundles/org.openhab.core.auth.oauth2client/src/main/java/org/openhab/core/auth/oauth2client/internal/OAuthClientServiceImpl.java

```diff
@@ -395,4 +397,17 @@ public boolean removeAccessTokenRefreshListener(AccessTokenRefreshListener liste
     private String createNewState() {
         return UUID.randomUUID().toString();
     }
+
+    @Override
+    public <T extends JsonDeserializer<?>> OAuthClientService withDeserializer(Class<T> deserializerClass) {
+        OAuthClientServiceImpl clientService = new OAuthClientServiceImpl(handle, persistedParams.tokenExpiresInSeconds,
```

Comment:
Yes, this looks nice. Do we have to create a new instance here or can we use `this`?

Reply:
I did opt to create a new instance to avoid any side effect regarding other parameter changed, it was the most straitforward path

Reply:
Understood. Sounds reasonable.